be/src/pipeline/exec/file_scan_operator.cpp

```diff
@@ -37,9 +38,11 @@ Status FileScanLocalState::_init_scanners(std::list<vectorized::VScannerSPtr>* s
     }
 
     auto& p = _parent->cast<FileScanOperatorX>();
-    uint32_t shard_num = std::min(
-            config::doris_scanner_thread_pool_thread_num / state()->query_parallel_instance_num(),
-            _max_scanners);
+    // There's only one scan range for each backend in batch split mode. Each backend only starts up one ScanNode instance.
+    uint32_t shard_num =
+            std::min(config::doris_scanner_thread_pool_thread_num /
+                             (_batch_split_mode ? 1 : state()->query_parallel_instance_num()),
```

Comment:
I suggest to extract this `_batch_split_mode ? 1 : state()->query_parallel_instance_num()` to a method in file scan operator.

Reply:
If extract this function to scan operator, It seems that the runtime state cannot be obtained through `state()` except through the parameters of this function. But this feels a bit strange.